In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import os

In [5]:
os.listdir()

['.git',
 '.ipynb_checkpoints',
 '.vscode',
 'dataset',
 'environment.yml',
 'heart_disease.ipynb']

In [11]:
CSV_PATH = 'dataset/heart.csv'
CSV_PATH = os.path.join(os.getcwd(), CSV_PATH)
heart_df = pd.read_csv(CSV_PATH)

In [12]:
heart_df

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     63    1   3       145   233    1        0      150      0      2.3   
1     37    1   2       130   250    0        1      187      0      3.5   
2     41    0   1       130   204    0        0      172      0      1.4   
3     56    1   1       120   236    0        1      178      0      0.8   
4     57    0   0       120   354    0        1      163      1      0.6   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
298   57    0   0       140   241    0        1      123      1      0.2   
299   45    1   3       110   264    0        1      132      0      1.2   
300   68    1   0       144   193    1        1      141      0      3.4   
301   57    1   0       130   131    0        1      115      1      1.2   
302   57    0   1       130   236    0        0      174      0      0.0   

     slope  ca  thal  target  
0        0   0     1       1  
1        0   0     2       1  
2        2   0     2       1  
3        2   0     2       1  
4        2   0     2       1  
..     ...  ..   ...     ...  
298      1   0     3       0  
299      1   0     3       0  
300      1   2     3       0  
301      1   1     3       0  
302      1   1     2       0  

[303 rows x 14 columns]

# Dataset parameters

1.  **Age**
2.	**Sex** (0 – female and 1 – male)
3.	**cp**: chest pain type

    - (1): typical angina
    - (2): atypical angina
    - (3): non-anginal pain
    - (4): asymptomatic
    
4.	**trestbps**: resting blood pressure (in mm Hg on admission to the hospital)
5.	**chol**: serum cholesterol in mg/dl
6.	 **fbs**: (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)
7.	**restecg**: resting electrocardiographic results

    - (0): normal
    - (1): having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)
    - (2): showing probable or definite left ventricular hypertrophy by Estes' criteria
    
8.	**thalach**: maximum heart rate achieved
9.	**exang**: exercise induced angina (1 = yes; 0 = no)
10.	**oldpeak** = ST depression induced by exercise relative to rest
11.	**slope**: the slope of the peak exercise ST segment

    - (1): upsloping
    - (2): flat
    - (3): downsloping
    
12.	**ca**: number of major vessels (0-3) colored by flourosopy
13.	**thal**: 3 = normal; 6 = fixed defect; 7 = reversable defect
14.	**target**: diagnosis of heart disease (angiographic disease status)

    - (0): < 50% diameter narrowing
    - (1): > 50% diameter narrowing (Predicted value)
    
